fulp_modules/main_features/bloodsuckers/code/antagonists/bloodsucker_datum.dm

```diff
@@ -535,42 +548,38 @@
 	lair_objective.generate_objective()
 	add_objective(lair_objective)
 
-	// Objective 1: Vassalize a Head/Command, or Drink X amount of Blood.
+	// Survive Objective
+	var/datum/objective/bloodsucker/survive/survive_objective = new
+	survive_objective.owner = owner
+	objectives += survive_objective
+
+	// Objective 1: Vassalize a Head/Command, or a specific target
 	switch(rand(0,1))
 		if(0) // Protege Objective
 			var/datum/objective/bloodsucker/protege/protege_objective = new
 			protege_objective.owner = owner
 			protege_objective.generate_objective()
 			protege_objective.objective_name = "Optional Objective"
-			add_objective(protege_objective)
-		if(1) // Drink Blood Objective
-			var/datum/objective/bloodsucker/gourmand/gourmand_objective = new
-			gourmand_objective.owner = owner
-			gourmand_objective.generate_objective()
-			gourmand_objective.objective_name = "Optional Objective"
-			add_objective(gourmand_objective)
+			objectives += protege_objective
+		if(1) // Vassalize Target Objective
+			var/datum/objective/bloodsucker/vassalhim/vassalhim_objective = new
+			vassalhim_objective.owner = owner
+			vassalhim_objective.objective_name = "Optional Objective"
+			objectives += vassalhim_objective
```

Comment:
forge_bloodsucker_objectives() is desperately in need of an update. It has an add_objective proc it uses once and then never again, and the code where it creates the datum, assigns the owner, generate the objective, and set the name is repeated so many times in the code

Condense this repeated functionality into add_objective

Reply:
#246 is removing add_objective

Reply:
In that case, you can add these functions to /datum/objective/bloodsucker/New() and condense it down to objectives += new (Whatever object procpath)